src/useChildrenActionHelper.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import {
+  NavigationAction,
+  NavigationState,
+  OnActionListener,
+  Router,
+} from './types';
+import NavigationBuilderContext from './NavigationBuilderContext';
+
+type Options = {
+  router: Router;
+  onAction: (action: NavigationAction, sourceNavigatorKey?: string) => boolean;
+  getState: () => NavigationState;
+  setState: (state: NavigationState) => void;
+  getKey: () => string | undefined;
+  childrenOnActionListeners: OnActionListener[];
+};
+
+export default function useChildrenActionHelper({
+  router,
+  onAction,
+  getState,
+  getKey,
+  setState,
+  childrenOnActionListeners,
+}: Options) {
+  const {
+    onChildUpdate: parentOnChildUpdate,
+    registerListener: parentRegisterListener,
+    unregisterListener: parentUnregisterListener,
+  } = React.useContext(NavigationBuilderContext);
+
+  const registerListener = (listener: OnActionListener) => {
+    childrenOnActionListeners.push(listener);
+  };
+  const unregisterListener = (listener: OnActionListener) => {
+    const index = childrenOnActionListeners.indexOf(listener);
+    childrenOnActionListeners.splice(index, 1);
+  };
+
+  React.useEffect(() => {
+    parentRegisterListener && parentRegisterListener(onAction);
+    return () => {
+      parentUnregisterListener && parentUnregisterListener(onAction);
+    };
+  }, [onAction, parentRegisterListener, parentUnregisterListener]);
```

Comment:
Need to memo `registerListener` and `registerListener`. Otherwise this effect will run every render in children as these functions are re-created every render.